components/rule-mgt/org.wso2.carbon.identity.rule.metadata/src/main/java/org/wso2/carbon/identity/rule/metadata/config/FieldDefinitionConfig.java

```diff
@@ -0,0 +1,120 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class to load field definitions from a file.
+ */
+public class FieldDefinitionConfig {
+
+    private final Map<String, FieldDefinition> fieldDefinitionMap;
+
+    private FieldDefinitionConfig(Map<String, FieldDefinition> fieldDefinitionMap) {
+
+        this.fieldDefinitionMap = fieldDefinitionMap;
+    }
+
+    public Map<String, FieldDefinition> getFieldDefinitionMap() {
+
+        return Collections.unmodifiableMap(fieldDefinitionMap);
+    }
+
+    public static FieldDefinitionConfig load(File file, OperatorConfig operatorConfig)
+            throws RuleMetadataConfigException {
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            TypeFactory typeFactory = mapper.getTypeFactory();
+            Map<String, Map<String, Object>> configMap =
+                    mapper.readValue(file, typeFactory.constructMapType(
+                            Map.class, String.class, Object.class));
+
+            Map<String, FieldDefinition> fieldDefinitionMap = new HashMap<>();
+
+            for (Map.Entry<String, Map<String, Object>> entry : configMap.entrySet()) {
+                String key = entry.getKey();
+                Map<String, Object> value = entry.getValue();
+
+                Field field = mapper.convertValue(value.get("field"), Field.class);
+                List<Operator> operators = loadOperators(value, operatorConfig, mapper);
+                Value valueMeta = convertValueObject(value, mapper);
+
+                FieldDefinition fieldDefinition = new FieldDefinition(field, operators, valueMeta);
+                fieldDefinitionMap.put(key, fieldDefinition);
+            }
+
+            return new FieldDefinitionConfig(fieldDefinitionMap);
+        } catch (Exception e) {
+            throw new RuleMetadataConfigException(
+                    "Error while loading field definitions from file: " + file.getAbsolutePath(), e);
+        }
+    }
+
+    private static List<Operator> loadOperators(Map<String, Object> value, OperatorConfig operatorConfig,
+                                                ObjectMapper mapper) {
+
+        List<Operator> operators = new ArrayList<>();
+        String[] operatorNames = mapper.convertValue(value.get("operators"), String[].class);
+        for (String operatorName : operatorNames) {
+            if (operatorConfig.getOperatorsMap().containsKey(operatorName)) {
+                operators.add(operatorConfig.getOperatorsMap().get(operatorName));
+            }
+        }
+        return operators;
+    }
+
+    private static Value convertValueObject(Map<String, Object> value, ObjectMapper mapper) {
+
+        Map<String, Object> valueObject =
+                mapper.convertValue(value.get("value"), new TypeReference<Map<String, Object>>() {
+                });
+
+        Value.InputType inputType = Value.InputType.valueOf(valueObject.get("inputType").toString().toUpperCase());
```

Comment:
`"inputType"` 
Shall we define these as constants in the class ?

Reply:
These stuff are not reused right ? My view is having the literal is more readable